packages/react-router/lib/rsc/server.rsc.ts

```diff
@@ -758,7 +779,11 @@ async function generateRenderResponse(
           }
         }
 
-        let staticContext = await query(request);
+        if (skipRevalidation) {
+          queryOptions.filterMatchesToLoad = () => false;
+        }
+
+        let staticContext = await query(request, queryOptions);
```

Comment:
I don't think we need a reference to `queryOptions` here since if you don't pass a new `filterMatchesToLoad` it'll use the existing one?  I think we could revert lines 719-725 above and simplify this to:

```suggestion
        let staticContext = await query(
          request,
          skipRevalidation ? { filterMatchesToLoad = () => false } : undefined
        );
```

or this would be functionally equivalent too?

```tsx
        let staticContext = await query(request, {
          filterMatchesToLoad: skipRevalidation ? () => false : undefined
        });
```